lib/wpc_proto/internal_modules/proto_otap.c

```diff
@@ -0,0 +1,115 @@
+/* Wirepas Oy licensed under Apache License, Version 2.0
+ *
+ * See file LICENSE for full license details.
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "wpc.h"
+#include "proto_otap.h"
+#include "proto_config.h"
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include "platform.h"
+#include "common.h"
+
+#define LOG_MODULE_NAME "otap_proto"
+#define MAX_LOG_LEVEL INFO_LOG_LEVEL
+#include "logger.h"
+
+static app_scratchpad_status_t otap_status = {0};
+
+static bool initialize_otap_variables()
+{
+    if (WPC_get_local_scratchpad_status(&otap_status) != APP_RES_OK)
+    {
+        LOGE("Cannot get local scratchpad status\n");
+        return false;
+    }
+    return true;
+}
+
+bool Proto_otap_init(void)
+{
+    /* Read initial config from sink */
+    if (!initialize_otap_variables())
+    {
+        LOGE("All the otap settings cannot be read\n");
```

Comment:
return false then?

Reply:
That's the question : do we want to fail init if we're not able to read some parts of the config ? In the last commit, it's moved to Proto_config_init(), where initialize_config_variables() doesn't prevent init to succeed.

Reply:
Decided to leave it as is : do not return false, values will be updated later.